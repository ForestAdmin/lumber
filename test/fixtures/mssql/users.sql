CREATE TABLE [dbo].users (
  id int GENERATED BY DEFAULT AS IDENTITY NOT NULL,
  username varchar(25) NOT NULL,
  enabled bit CONSTRAINT defo_users_enabled DEFAULT true,
  last_login timestamp NOT NULL CONSTRAINT defo_users_last_login DEFAULT now(),
  CONSTRAINT users_pkey PRIMARY KEY (id)
);