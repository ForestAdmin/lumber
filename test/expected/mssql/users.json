{
  "fields":[
    {
      "name":"username",
      "nameColumn":"username",
      "type":"STRING",
      "primaryKey":false,
      "defaultValue":null
    },
    {
      "name":"enabled",
      "nameColumn":"enabled",
      "type":"BOOLEAN",
      "primaryKey":false,
      "defaultValue":true
    },
    {
      "name":"lastLogin",
      "nameColumn":"last_login",
      "type":"DATE",
      "primaryKey":false,
      "defaultValue":"now()"
    }
  ],
  "references":[
    {
      "isHasOneOrHasMany":true,
      "ref":"addresses",
      "association":"hasOne",
      "foreignKeyName":"undefinedKey"
    },
    {
      "isHasOneOrHasMany":true,
      "ref":"reviews",
      "association":"hasMany",
      "foreignKeyName":"undefinedKey"
    },
    {
      "isBelongsToMany":true,
      "ref":"users_books",
      "association":"belongsToMany",
      "junctionTable":"books",
      "foreignKeyName":"undefinedKey"
    }
  ],
  "primaryKeys":["id"],
  "options":{
    "underscored":true,
    "timestamps":false,
    "hasIdColumn":true,
    "hasPrimaryKeys":true
  }
}